tst/scripts/utils/artemis.py

```diff
@@ -43,9 +43,13 @@
 def set_executable(executable_path):
     global artemis_executable
     global artemis_run_dir
+    global artemis_fig_dir
+    global artemis_log_dir
     global custom_exe
     artemis_executable = executable_path
     artemis_run_dir = os.path.join(os.path.dirname(artemis_executable), "tst")
+    artemis_fig_dir = os.path.join(current_dir, "figs")
+    artemis_log_dir = current_dir
```

Comment:
Does this work with the `--exe` flag? Suppose I'm in `~/github/artemis` and to `tst/run_tests.py --exe /scratch/myname/mydir/myexecutable`, I would want figs and logs to appear in `/scratch/myname/mydir/` but this will write them to `~/github/artemis` right?

Reply:
I think it makes sense to launch the tests from the directory you want the results in. For me, that's the top of a build directory. For normal testing that would be in `tst/`. For `--exe`, I guess all bets are off? What do you want this to be for that case? `artemis_run_dir`?

Reply:
Say we made a release on a supported system in `/usr/projects/artemis/releases/..`. If a user wanted to run the tests on that release, they would pass that path to `--exe`, but where should all the outputs and logs go? I would think they shouldn't go in the release folder where the exe is, right? They should go in some local directory of the user.

Reply:
Yeah thats a good point, and I think the same issue would appear with `cd /usr/projects/artemis/releases && python ~/artemis/tst/run_tests.py`. What do you think about default output being exec dir for `--exe` and current dir for `run_test.py` when it finds the executable (same behavior as now) but an optional `--output_dir` that replaces `--log_file` and gives the dir both the logfile and the figs outputs will be written to (and just always create a logfile)?

Reply:
In my example, the user shouldn't be running in `/usr/projects/artemis/releases`. No output should be written there. All of their output, including snapshots, should be somewhere else (I guess in your `--output_dir`).

Reply:
OK agreed. Yeah and hopefully permissions would be set up that, with the `--exe` flag pointing to an executable in `release`, the user couldn't write to that directory and they would need to fix the issue with e.g. `--output_dir`. 

I would at least still like to update this PR so the default `--exe [path]` behavior remains writing the log/fig output to a `tst` dir in the same directory as the `--exe`. If it fails in your `releases` example for now due to permissions that's fine. I think it is just a matter of updating `artemis_fig_dir` and `artemis_log_dir` when calling `set_executable` only for the `if artemis_exe_path is not None` codepath in `run_tests.py`, so `set_executable` might need an additional boolean flag like `update_output_dirs`.

Reply:
Should be good now. I added `--output_dir`. That will be the base if `--exe` is not set. If you include the build stage, then `--output_dir` is ignored and you get the default behavior in `artemis.py`.